tapeagents/mcp.py

```diff
@@ -104,6 +109,23 @@ async def connect_to_server(self, server_name: str, server_params: StdioServerPa
 
     async def call_tool(self, tool_name: str, tool_args: dict[str, Any]) -> CallToolResult:
         server_name = self.check_tool_exists(tool_name)
+        # Current implementation of cache assumes tool calls are deterministic and do not alter state
+        if self.use_cache:
+            tool_name_key = f"mcp_{tool_name}"
```

Comment:
lets also include the server name into the key, to avoid issues when new config with another server uses the same fn name. mcp_servername_toolname